vision_datasets/common/data_manifest.py

```diff
@@ -96,14 +98,30 @@ def __init__(self, id, img_path, width, height, labels):
                 classification: [c_id] for multiclass, [c_id1, c_id2, ...] for multilabel;
                 detection: [[c_id, left, top, right, bottom], ...];
                 image caption: [caption1, caption2, ...];
-                image_text_matching: [(text1, match (0 or 1), text2, match (0 or 1), ...)]
+                image_text_matching: [(text1, match (0 or 1), text2, match (0 or 1), ...)];
                 multitask: dict[task, labels]
+            label_file_path (str): path to a image label file. "label_file_path" only works for image matting task.
         """
         self.id = id
         self.img_path = img_path
         self.width = width
         self.height = height
-        self.labels = labels
+        self._labels = labels
+        self.label_file_path = label_file_path
+        self.file_reader = FileReader()
+
+    @property
+    def labels(self):
+        if self.label_file_path:
+            label = Image.open(self.file_reader.open(self.label_file_path))
+            self.file_reader.close()
```

Comment:
My previous suggestion might not be ideal. Let us just do

with FileReader() as file_reader:
    label = Image.open(file_reader.open(self.label_filepath))

Reply:
with a class object as xxx is not permitted.

Reply:
> this breaks the next attempt to read the file

why this will breaks the next attempt to read the file? I think every time we are calling self.file_reader.open() to read the file

Reply:
The image has been converted to numpy array now, so it can directly read from the numpy array, no need to load the image data from the file.

Reply:
looks like file_reader is still used which makes it necessary to close it